### ISSUE TITLE
BatchJobHelper.downloadBatchJobMutateResponse fails with: No deserializer for {https://adwords.google.com/api/adwords/cm/v201605}ErrorList

### ISSUE THREAD

**Issue by jradcliff** [googleads/googleads-java-lib#89]
On certain HotSpot JVMs, downloading BatchJob results fails with the following error:

Caused by: org.xml.sax.SAXException: No deserializer for {https://adwords.google.com/api/adwords/cm/v201605}ErrorList
        at org.apache.axis.encoding.ser.BeanDeserializer.onStartChild(BeanDeserializer.java:314)
        at org.apache.axis.encoding.DeserializationContext.startElement(DeserializationContext.java:1035)
        at org.apache.axis.message.SAX2EventRecorder.replay(SAX2EventRecorder.java:165)
        at org.apache.axis.message.MessageElement.publishToHandler(MessageElement.java:1141)
        at org.apache.axis.message.MessageElement.getValueAsType(MessageElement.java:917)
        at com.google.api.ads.adwords.axis.utils.AxisDeserializer.deserializeBatchJobMutateResults(AxisDeserializer.java:112)
        at com.google.api.ads.adwords.axis.utils.v201605.batchjob.BatchJobHelperImpl.downloadBatchJobMutateResponse(BatchJobHelperImpl.java:104)

**Comment by jradcliff** [comment]
The `MutateResult` and `ErrorList` types in the `batchjob` package have properties defined in the `BatchJobMutateResultInterface` and `BatchJobErrorListInterface` interfaces, respectively.

On some Hotspot JVMs, if [java.beans.Introspector](https://docs.oracle.com/javase/7/docs/api/java/beans/Introspector.html) (used by Axis) encounters a property defined via an implemented interface, it will return a `PropertyDescriptor` where `getPropertType()` returns the *interface* class instead of the type declared in the implementing class (e.g., `BatchJobErrorListInterface` instead of `ErrorList`).

This causes problems during Axis deserialization because Axis relies on the presence of a static `getTypeDesc` method on each property it encounters. Since that method does not exist on the *interface*, Axis fails to find the correct type descriptor for the attribute.

For example, `MutateResult` has an `errorList` property defined as:

```
  @Override
  public ErrorList getErrorList() {
    return errorList;
  }
```

This implements the following method from `BatchJobMutateResultInterface` interface:

```
  public BatchJobErrorListInterface<E> getErrorList();
```

In OpenJDK JVMs, `PropertyDescriptor.getPropertyType()` will return the `ErrorList` class.

On some HotSpot JVMs, `PropertyDescriptor.getPropertyType()` will return the `BatchJobMutateResultInterface$BatchJobErrorListInterface` class.

**Comment by jradcliff** [comment]
Fixed in [2.16.0](https://github.com/googleads/googleads-java-lib/releases/tag/2.16.0).

**Comment by jradcliff** [comment]
For posterity, the related HotSpot bugs are:

*  [JDK-8067267](https://bugs.openjdk.java.net/browse/JDK-8067267)
*  [JDK-8027905](https://bugs.openjdk.java.net/browse/JDK-8027905)
*  [JDK-7172854](https://bugs.openjdk.java.net/browse/JDK-7172854)
*  [JDK-8041555](https://bugs.openjdk.java.net/browse/JDK-8041555)
